src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -259,11 +258,12 @@ private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher
             }
         }
     }
-
-    public String constructCommandForTest(String inputArguments) {
-        final String matlabFunctionName =
-                FilenameUtils.removeExtension(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE);
-        final String runCommand = "exit(" + matlabFunctionName + "(" + inputArguments + "))";
+    
+    public String constructCommandForTest(String inputArguments, FilePath scriptPath) {
+        final String matlabFunctionName = MatlabBuilderConstants.MATLAB_TEST_RUNNER_FILE_PREFIX
+                + scriptPath.getBaseName().replaceAll("-", "_");
+        final String runCommand = "addpath('" + scriptPath.getRemote().replaceAll("'", "''")

```

Comment:
By using `addpath`, the temp path is going to be on the user's path throughout the entire run. We try to avoid modifying the path as much as possible. Consider using `cd` to go to the folder containing the script and then `cd` back in the script. I think that's how you are handling this in Run MATLAB Command.

Reply:
I have to ad path anyways in the script file to get testScript.writeToText() and run(testScript) recognized. Changed it accordingly.